sdk/core/Azure.Core/src/RequestUriBuilder.cs

```diff
@@ -185,15 +201,21 @@ public void AppendQuery(string name, string value, bool escapeValue)
             {
                 _pathAndQuery.Append('&');
             }
-
+#if NETCOREAPP2_1_OR_GREATER
             _pathAndQuery.Append(name);
+#else
+            _pathAndQuery.Append(name.ToString());
+#endif
             _pathAndQuery.Append('=');
-            if (escapeValue && !string.IsNullOrEmpty(value))
+            if (escapeValue && !value.IsEmpty)
             {
-                value = Uri.EscapeDataString(value);
+                value = Uri.EscapeDataString(value.ToString()).AsSpan();
```

Comment:
It looks like we're calling `value.ToString()` multiple times... would it make sense to cache the value if perf is the concern?